src/main/java/rainy/tasks/Task.java

```diff
@@ -0,0 +1,77 @@
+package rainy.tasks;
+
+import java.time.LocalDate;
+
+import rainy.database.UI;
+
+import rainy.rainyexceptions.InvalidMarkAndUnmarkException;
+
+/**
+ * Represents each individual task. Its parameters tells users of details such as its completion status,
+ * name, date, and a <code>UI</code> object to display chatbot messages.
+ */
+public class Task implements Comparable<Task> {
+    private boolean isDone;
```

Comment:
I like how you make sure that boolean variable name has 'is' in front (i.e. using 'isDone' instead of 'done')